clients/src/test/java/org/apache/kafka/common/utils/ChunkedBytesStreamTest.java

```diff
@@ -149,6 +160,38 @@ public void testCorrectnessForMethodSkip(int bytesToPreRead, ByteBuffer inputBuf
         }
     }
 
+    @Test
+    public void testEdgeCaseInputForMethodSkip() throws IOException {
+        int bufferLength = 16;
+        ByteBuffer inputBuf = ByteBuffer.allocate(bufferLength);
+        RANDOM.nextBytes(inputBuf.array());
+        inputBuf.position(inputBuf.capacity());
+        inputBuf.flip();
+
+        try (InputStream is = new ChunkedBytesStream(new ByteBufferInputStream(inputBuf.duplicate()), supplier, 10, true)) {
```

Comment:
we want to test this for both cases when skip is pushed down vs not, i.e. last argument is true and when last argument is false.

Reply:
you can use `@ParameterizedTest` for true/false here instead of duplicating code.